05.wc/wc.rb

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def main
+  options = ARGV.getopts('lwc')
+  options.transform_values!(&:!) if options.values.none?
+  filenames = ARGV
+
+  if filenames.any?
+    handle_multiple_files(filenames, options)
+  else
+    text = readlines.join
+    handle_standard_input(text, options)
+  end
+end
+
+def handle_standard_input(text, options)
+  detail_hashes = create_detail_hash(text)
+  display_values([detail_hashes], options)
+end
+
+def handle_multiple_files(files, options)
+  detail_hashes = files.map do |file|
+    text = File.read(file)
+    detail = create_detail_hash(text)
+    detail['name'] = file
+    detail
+  end
+  if files.length >= 2
+    total = { 'name' => 'total' }
+    options.each_key do |option|
+      total[option] = detail_hashes.sum { |hash| hash[option] }
+    end
+    detail_hashes << total
+  end
+  width = calc_column_width(detail_hashes, options)
+  display_values(detail_hashes, options, width)
+end
+
+def calc_column_width(detail_hashes, options)
+  options_count = options.count do |_key, value|
+    value == true
+  end
```

Comment:
```suggestion
  options_count = options.count { |_k, v| v }
```

vが真のカウントを取得すればいいので、`== true` はなくせるのと、これぐらいなら1行で書いても問題ないと思います